ssl/s3_enc.c

```diff
@@ -155,6 +155,7 @@ int ssl3_change_cipher_state(SSL *s, int which)
         RECORD_LAYER_reset_read_sequence(&s->rlayer);
         mac_secret = &(s->s3->read_mac_secret[0]);
     } else {
+        s->statem.invalid_enc_write_ctx = 1;
```

Comment:
Perhaps for the intervening SSLfatal calls (where we know that no alert will be sent), we should set the alert code to SSL_AD_NO_ALERT.

Reply:
Yes, I was also thinking about that possiblilty.  I had the two reasons against it:
- it would be okay to send an alert when the same happens with the enc_read_ctx.
  but called functions don't know which one they operate on.
- in the case there is an error, the change_cipher_state is left with still invalid enc_write_ctx,
  so that must still be prevented, otherwise we crash afterwards , or even worse send
  something unencrypted.

Reply:
Yes, that makes sense.